mkt/webapps/fakedata.py

```diff
@@ -0,0 +1,263 @@
+    generate_hosted_manifest(a)
+    return a
+
+
+def generate_hosted_manifest(app):
+    data = {
+        'name': unicode(app.name),
+        'description': 'This app has been automatically generated',
+        'version': '1.0',
+        'icons': {
+            '16': 'http://testmanifest.com/icon-16.png',
+            '48': 'http://testmanifest.com/icon-48.png',
+            '128': 'http://testmanifest.com/icon-128.png'
+        },
+        'installs_allowed_from': ['*'],
+        'developer': {
+            'name': 'Marketplace Team',
+            'url': 'https://marketplace.firefox.com/credits'
+        }
+    }
+    AppManifest.objects.create(
+        version=app._latest_version, manifest=json.dumps(data))
+    app.update(manifest_url='http://%s.testmanifest.com/manifest.webapp' %
+               (slugify(unicode(app.name)),))
+
+
+def generate_app_package(app, out, apptype, permissions, version='1.0',
+                         num_locales=2):
+    manifest = {
+        'version': version.version,
+        'name': unicode(app.name),
+        'description': ('This packaged app has been automatically generated'
+                        ' (version %s)' % (version.version,)),
+        'icons': {
+            '16': '/icons/16.png',
+            '32': '/icons/32.png',
+            '256': '/icons/256.png'
+        },
+        'developer': {
+            'name': 'Marketplace Team',
+            'url': 'https://marketplace.firefox.com/credits'
+        },
+        'installs_allowed_launch': ['*'],
+        'from_path': 'index.html',
+        'locales':  dict((lang, {
+            'name': name,
+            'description': 'This packaged app has been automatically generated'
+        }) for lang, name in generate_localized_names(
+            app.name, num_locales).items()),
+        'permissions': dict(((k, {"description": k})
+                             for k in permissions)),
+        'default_locale': 'en',
+        'orientation': 'landscape',
+        'type': apptype,
+        'fullscreen': 'true'
+    }
+    outz = ZipFile(file=out, mode='w', compression=ZIP_DEFLATED)
+    try:
+        for size in ('32', 'med'):
+            outz.writestr(
+                'icons/%s.png' % (size,),
+                open(os.path.join(
+                    settings.MEDIA_ROOT,
+                    'img/app-icons/%s/generic.png' % (size,))).read())
+        outz.writestr('script.js',
+                      'document.onload=function() {alert("Hello!");};')
+        outz.writestr(
+            'index.html',
+            '<title>Packaged app</title><script src="script.js"></script>'
+            '<h1>Test packaged app</h1>')
+        outz.writestr("manifest.webapp", json.dumps(manifest))
+    finally:
+        outz.close()
+    AppManifest.objects.create(
+        version=version, manifest=json.dumps(manifest))
+
+
+
+def generate_packaged_app(name, apptype, category, permissions=(),
+                          versions=(4,), num_locales=2, **kw):
```

Comment:
/me looks suspiciously at data structures as default args. I guess that's fine b/c tuples are immutable.